feat(python): Adding CDK example chaining S3, SNS, SQS and Lambda (#521)
aws-samples/aws-cdk-examples: python/s3-sns-sqs-lambda-chain/README.md

# Chaining S3 to SNS to SQS to Lambda
---

<!--BEGIN STABILITY BANNER-->
---

![Stability: Stable](https://img.shields.io/badge/stability-Stable-success.svg?style=for-the-badge)

> **This is a stable example. It should successfully build out of the box**
>
> This example is built on Construct Libraries marked "Stable" and does not have any infrastructure prerequisites to build.
---
<!--END STABILITY BANNER-->

## Overview

> A common pattern that I have used to create a resilient way to trigger a lambda when an event occurs on S3 is to chain the following services:

Reviewer: ```suggestion
Create a resilient way to trigger a lambda when an event occurs on S3. It chains the following services:
```

virtualenv fails, you can create the virtualenv manually.

To manually create a virtualenv on MacOS and Linux:

```
$ python3 -m venv .env
```

After the init process completes and the virtualenv is created, you can use the following
step to activate your virtualenv.

```
$ source .env/bin/activate
```

If you are a Windows platform, you would activate the virtualenv like this:

```
% .env\Scripts\activate.bat
```

Once the virtualenv is activated, you can install the required dependencies.

```
$ pip install -r requirements.txt
```

At this point you can now synthesize the CloudFormation template for this code.

```
$ cdk synth
```

### CDK Deploy

When deployed, CDK creates a layer .zip
asset to be stored in a staging bucket managed by CDK. To enable this the AWS account being used needs to be
[bootstrapped](https://docs.aws.amazon.com/cdk/latest/guide/bootstrapping.html).

With default profile,
```
$ cdk bootstrap
```

With specific profile,
```
$ cdk bootstrap --profile test
```

With the bootstrap complete, we are ready to deploy the lambda function and lambda layer.

```
$ cdk deploy

> ```

Reviewer: Would you mind adding some documentation on how to run the tests? 🙏🏻

Author: Sure thing, I will add that shortly.
